src/array/array.jl

```diff
@@ -429,6 +430,13 @@ function DimArray(A::AbstractBasicDimArray;
     newdata = collect(data)
     DimArray(newdata, format(dims, newdata); refdims, name, metadata)
 end
+# Write a single column from a table with one or more coordinate columns to a DimArray
+function DimArray(table, dims; name=NoName(), selector=DimensionalData.Near(), precision=6, kw...)
+    data = restore_array(table, dims; selector=selector, missingval=missing, name=name, precision=precision)
```

Comment:
We should probably do a `Tables.istable` check at this point for a nicer error and stack trace, rather than failing deep in the internals.